src/frontend/components/sidebar.tsx

```diff
@@ -1,14 +1,55 @@
-import React from 'react'
-import MenuOptions from './menu-options'
+"use client"
+import React, { useEffect, useState } from 'react';
+import MenuOptions from './menu-options';
+import { IoIosArrowForward } from "react-icons/io";
 
-type Props = {}
+type Props = {
+  isOpen: boolean;
+  toggleSidebar: () => void;
+};
 
 const Sidebar = (props: Props) => {
-    return (
-        <aside className="sticky top-0 overflow-y-auto h-full bg-primary text-white py-1 px-2 w-64 md:block hidden">
-            <MenuOptions />
+  const [sidebarOpen, setSideBarOpen] = useState(true);
+
+  const handleViewSidebar = () => {
+    setSideBarOpen(!sidebarOpen);
+  };
+
+  useEffect(() => {
+    const handleResize = () => {
+      if (window.innerWidth >= 768) {
+        setSideBarOpen(true);
+      } else {
+        setSideBarOpen(false);
+      }
+    };
+
+    handleResize(); // Set initial state based on screen size
+    window.addEventListener('resize', handleResize);
+    return () => window.removeEventListener('resize', handleResize);
+  }, []);
+
+  return (
+    <div>
+      <aside
+        className={`fixed top-0 h-full bg-primary text-white py-1 px-2 w-64 transition-transform duration-300 ease-in-out ${
+          sidebarOpen ? 'translate-x-0' : '-translate-x-full'
+        }`}
+      >
+        <MenuOptions isOpen={sidebarOpen} toggleSidebar={handleViewSidebar} />
+      </aside>
+      {!sidebarOpen && (
+        <aside className="h-full w-full flex items-center justify-center bg-transparent">
```

Comment:
This component occupies viewport space and shifts the entire <main>, which is particularly problematic on smaller screens. Consider moving it as an overlay or placing the button in the top bar next to the product title.

Reply:
Done 
![Screenshot from 2024-07-19 21-06-57](https://github.com/user-attachments/assets/1ce9af14-c822-413b-868f-f27347ed286a)